server/fleet/hosts.go

```diff
@@ -387,6 +387,54 @@ type Host struct {
 	Policies *[]*HostPolicy `json:"policies,omitempty" csv:"-"`
 }
 
+type HostForeignVitalGroup struct {
+	Name  string
+	Query string
+}
+
+type HostVitalType int
+
+const (
+	HostVitalTypeDomestic HostVitalType = iota // Domestic vitals are those that are stored in the host table
+	HostVitalTypeForeign                       // Foreign vitals are those that are stored in a separate table and joined to the host table
+	HostVitalTypeCustom                        // Custom vitals are those that are stored in the host_additional table as a JSON blob
+)
+
+type HostVital struct {
+	Name              string // Display name of the vital
+	VitalType         HostVitalType
+	DataType          string  // Data type of the vital, e.g. "string", "int", "bool"
+	ForeignVitalGroup *string // For foreign vitals, the group they belong to
+	Path              string  // Path to the vital in the SQL query, for use in generating the WHERE clause
+}
+
+var hostForeignVitalGroups = map[string]HostForeignVitalGroup{
+	"idp": {
+		Name:  "Identity Provider",
+		Query: `RIGHT JOIN host_scim_user ON (hosts.id = host_scim_user.host_id) JOIN scim_users ON (host_scim_user.scim_user_id = scim_users.id) JOIN scim_user_group ON (host_scim_user.scim_user_id = scim_user_group.scim_user_id) JOIN scim_groups ON (scim_user_group.group_id = scim_groups.id)`,
+	},
+}
+
+var hostVitals = map[string]HostVital{
+	"end_user_idp_groups": {
```

Comment:
I coded the frontend to use `...group`, no `s`, based on [this](https://github.com/fleetdm/fleet/pull/30015/files#diff-7246bc304b15c8865ed8eaa205e9c244d0a0314e4bae60cf553dc06147c38b64R4851) and [this](https://github.com/fleetdm/fleet/pull/30015/files#r2150569636), but I see the example [here](https://github.com/fleetdm/fleet/pull/30015/files#r2150549630) has the plural, probably the reason for the mismatch.

These will be specified one at a time right, so if looking at multiple groups it would be multiple `HostVital`s? So `group` no plural makes sense?

Either way

Reply:
yeah makes sense to me, i'll update